eip-0028.md

```diff
@@ -0,0 +1,130 @@
+- [Implementation in Mobile Wallet](#implementation-in-mobile-wallet)
+- [Implementation in dApp](#implementation-in-dapp)
+- [Benefits for dApps](#benefits-for-dapps)
+- [Benefits for Mobile Wallets](#benefits-for-mobile-wallets)
+
+## Description
+This EIP defines a standard for trustless authentication of users of a wallet app and an online
+dApp.
+
+## Background And Motivation
+
+dApps might want to validate if dApp users are really who they pretend to be. This is especially
+useful for dApps that grant certain abilities to holders of special tokens. At the moment,
+proofing that a user owns a token can only be done by sending the token to a depositary address.
+By sending the token, the user proofed to have access privileges to the token.
+However, sending token around is not always desirable. Especially for valuable tokens, users might
+not want to send it away, and doing two transactions (one to send it to the depositary address, one
+to refund it back) costs both time and transaction fees.
+
+To overcome this, ErgoAuth proposes a way to authenticate users trustless to have access to certain
+addresses storing a box. The protocol is trustless in both ways: The users don't need to trust
+the dApp, because the dApp does not get access to funds or secrets. The dApp don't need to trust
+the users or the wallet app, because it can validate the authentication keys.
+
+## ErgoAuth authentication protocol
+
+An authentication with ErgoAuth is driven by a dApp that needs to authenticate a user.
+
+1) The user enters the necessary information in the dApp's UI for the dApp to know if authentication
+is necessary. For example, users might enter their P2PK address (or, instead of manually entering,
+use ErgoPay to send the address to the dApp automatically).
+
+2) The dApp determines that authenticating the user is needed. For this, the dApp prepares a unique
+message that the wallet app should sign with a user's private key, and an Ergo address that the user
+needs to authenticate for. This might be a P2PK address, but in general P2S addresses can be also
+used for authentication.
+
+3) The dApp presents an ErgoAuth link for the user to click and open the wallet app and a QR code
+for mobile users to scan from within the wallet app.
+
+4) The wallet application parses the QR code/link data and obtains a
+`ErgoAuthRequestUrl` to fetch the actual `ErgoAuthRequest` data from
+(see [Data Formats](#data-formats) section).
+
+5) When `ErgoAuthRequest` is obtained, the wallet presents a screen showing that a dApp wants to
+authenticate the user, and the address the request is for. The wallet app should also inform the
+user that no funds or moved and no secrets will leave the device.
+
+6) When the user agrees, the wallet app signs the obtained message from ErgoAuthRequest and sends
+the signed message to the ErgoAuthRequest's replyToUrl.
+
+7) The dApp validates the signed message. When successful, it can proceed with its flow.
```

Comment:
An extension of the workflow may be specified which uses Cold Wallet for signing.

Reply:
For the moment, cold wallets won't work with ErgoAuth because the interchange format between a cold and a hot wallet can only transfer transaction signing requests for now. I think instead of extending it here, we need to extend the Cold Wallet EIP later.

Reply:
Yes, here you can refer to the next Cold Wallet EIP revision (mentioning it is not yet implemented).
Will serve as an advertisement for this EIP.